mle/agents/clarifier.py

```diff
@@ -0,0 +1,76 @@
+from rich.console import Console
+
+from mle.function import *
+
+
+class ClarifierAgent:
+
+    def __init__(self, model, console=None):
+        """
+        ClarifierAgent: An agent to enhance the understanding of user needs through multiple rounds of
+        Clarification Dialogue, which can improve the subsequent task generation effectiveness.
+
+        Args:
+            model: the model to use.
+        """
+        self.model = model
+        self.chat_history = []
+
+        self.console = console
+        if not self.console:
+            self.console = Console()
+
+        self.sys_prompt = f"""
+        You are an Machine learning Product Manager, you are going to collaborate with the user to plan the ML
+        project. You need to ensure that the user's ML project has no ambiguities, for example, in the use of
+        the dataset or the project requirements.
+
+        You have the capability to interact with the user by invoking the following tools, thereby enhancing
+        your understanding of the ML project.
+
+        1. Ask a multiple-choice question to the user by using function `ask_choices`.
+        2. Ask a yes/no question to the user using function `ask_yes_no`.
+        3. Ask a question to the user and get a response using function `ask_question`.
+        
+        If you can ensure there is no ambiguities in project understanding, output a short final summary.
+        """
+
+        self.functions = [
+            schema_ask_choices,
+            schema_ask_question,
+            schema_ask_yes_no,
+        ]
+
+        self.chat_history.append({"role": "system", "content": self.sys_prompt})
+
+    def interact(self, input: str, type: str):
+        """
+        Iteract with users to enhance the understanding of user needs
+        Args:
+            input (str): the project description.
+            type (str): the type of the project description, 'dataset' or 'requirement'.
+        """
+        if type == "dataset":
+            user_prompt = f"""
+            The dataset description provided by the user for this project is: `{input}`
+            If the dataset description is not a specific dataset or a valid file path, you can supply some publicly
+            available datasets for the user to choose from. You can only ask the user within the calling functions no
+            more than once. Finally, you can only output the final dataset name or path.
+            """
+        else:
+            user_prompt = f"""
```

Comment:
lets remove this requirement part

Reply:
just `dataset` is good enough for a quick start

Reply:
Yeah, the user requirement doesn't require many interactions (we have tried that before, and it's not a good workflow). 

And the interaction functions should not be called by the function calls -- they have bugs : (

Reply:
Okay, it seems that interactions that are too long do indeed reduce user interest in using our tool. :)